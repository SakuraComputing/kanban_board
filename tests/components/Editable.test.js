import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Editable from '../../app/components/Editable';


describe('Editable Component', () => {
  let wrapper, editing, value, onValueClick,
    onEdit, onDelete, defaultValue, renderDelete,
    deleteButton;


  beforeEach(() => {

    editing = jest.fn();
    value = jest.fn();
    onValueClick =  jest.fn();
    onEdit = jest.fn();
    onDelete = jest.fn();
    defaultValue = jest.fn();
    renderDelete = jest.fn();

    wrapper = shallow(<Editable
      editing={editing}
      value={value}
      onValueClick={onValueClick}
      onEdit={onEdit}
      onDelete={onDelete}
    />);
    deleteButton = <button className="delete" onClick={onDelete}>x</button>;
  });

  it('should exist', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should render the form correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should return input on renderEdit', () => {
    const anonymous = () => ({ baz: foo2, boz: 1 });
    const returnInput = wrapper.instance().renderEdit();
    const test = <input
      autoFocus={true}
      defaultValue={defaultValue}
      onBlur={anonymous}
      onKeyPress={anonymous}
      type="text"
    />;
    expect(toJSON(returnInput)).toEqual(toJSON(test));
  });
  it('should display delete button when renderDelete is called', () => {
    const returnDelButton = wrapper.instance().renderDelete();
    expect(returnDelButton).toEqual(deleteButton);
  });
  it('should render value when renderValue is called', () => {
    const retValue = wrapper.instance().renderValue();
    const test = <div onClick={onValueClick}>
      <span className="value">{defaultValue}</span>
      {deleteButton}
    </div>;
    expect(toJSON(retValue)).toEqual(toJSON(test));
  });
});
