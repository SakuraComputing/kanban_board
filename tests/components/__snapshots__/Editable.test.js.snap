// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editable Component should render the form correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editable
    editing={[MockFunction]}
    onDelete={[MockFunction]}
    onEdit={[MockFunction]}
    onValueClick={[MockFunction]}
    value={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        autoFocus={true}
        defaultValue={[MockFunction]}
        onBlur={[Function]}
        onKeyPress={[Function]}
        type="text"
      />,
      "onDelete": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoFocus": true,
        "defaultValue": [MockFunction],
        "onBlur": [Function],
        "onKeyPress": [Function],
        "type": "text",
      },
      "ref": [Function],
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          autoFocus={true}
          defaultValue={[MockFunction]}
          onBlur={[Function]}
          onKeyPress={[Function]}
          type="text"
        />,
        "onDelete": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": true,
          "defaultValue": [MockFunction],
          "onBlur": [Function],
          "onKeyPress": [Function],
          "type": "text",
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
